libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -693,10 +693,15 @@ void IRGeneratorForStatements::endVisit(UnaryOperation const& _unaryOperation)
 		else
 			solUnimplementedAssert(false, "Unary operator not yet implemented");
 	}
+	else if (resultType.category() == Type::Category::FixedBytes)
+	{
+		solAssert(op == Token::BitNot, "");
```

Comment:
Please add an assertion like this: https://github.com/ethereum/solidity/pull/10410/files#diff-8e1a453b201906ae5f79d442a51974453a9a434e6631837f9b54f1c8b2781c67R658

Reply:
The link doesn't seem to work, but I assume you refer to the one in bool case below?

Reply:
check that the input and the result type ar ethe same, as in the integer case.